### ISSUE TITLE
Bug: mockStatic don't unmock

### ISSUE THREAD

**Issue by samirda** [mockk/mockk#99]
### Expected Behavior

Return real value when not mocked

### Current Behavior

Returns mocked value for calls when mocking is done in other method.
Returns mocked value when mocking is done in other class

#### Mockk 1.8.4
```kotlin
class Temp1 {
	@Test
	fun someTest() {
		val timestamp = Instant.now().toEpochMilli()
		assertNotEquals(123L, timestamp)
	}

	@Test
	fun someOtherTest() {
		mockkStatic(Instant::class)
		every { Instant.now().toEpochMilli() } returns 123

		val timestamp = Instant.now().toEpochMilli()
		assertEquals(123, timestamp)
		
		unmockkStatic(Instant::class)
	}
}
```

```kotlin
class Temp2 {
	@Test
	fun someTest() {
		val timestamp = Instant.now().toEpochMilli()
		assertNotEquals(123L, timestamp)
	}
}
```

**Comment by oleksiyp** [comment]
Thanks for reporting it

**Comment by samirda** [comment]
Thanks.

**Comment by serandel** [comment]
For me it's happening again in 1.10.5 :(

**Comment by RubenGonzalezGonzalez** [comment]
I have the same problem in 1.10.5, and I tested the last published version (1.12.1) and it's still happening.
